src/Receive/ReceiveScreen.tsx

```diff
@@ -2,33 +2,26 @@ import _ from 'lodash'
 import React from 'react'
 import {defineMessages, useIntl} from 'react-intl'
 import {ActivityIndicator, ScrollView, StyleSheet, View} from 'react-native'
-import {useDispatch} from 'react-redux'
 
 import {Button, OfflineBanner, Spacer, StatusBar} from '../components'
-import {generateNewReceiveAddress, generateNewReceiveAddressIfNeeded} from '../legacy/actions'
+import {useWallet} from '../hooks'
 import {useSelectedWallet} from '../SelectedWallet'
 import {COLORS} from '../theme'
+import {YoroiWallet} from '../yoroi-wallets'
 import {AddressDetail} from './AddressDetail'
 import {UnusedAddresses, UsedAddresses} from './Addresses'
 
 export const ReceiveScreen = () => {
   const strings = useStrings()
   const wallet = useSelectedWallet()
-  const receiveAddresses = wallet.externalAddresses.slice(0, wallet.numReceiveAddresses)
```

Comment:
Bug. Component not aware of addresses changes. Used to fix: https://github.com/Emurgo/yoroi-mobile/pull/2209/files#diff-c94a2ef1655f7c1de12a45bf67350d7a4ab1c6d8381ae79cc03441ed7d1854b6R64